### PR TITLE
Make Selection visible everywhere

### PR DESCRIPTION
**Pull request by lainsce**
This target `*:selected` should capture all selected states and as bonus help with some broken UI states currently present with the stylesheet.

It's overridable by using the `&:selected` block on the widget that needs to be overridden.

### CODE REVIEW

**Comment by danrabbit** [comment]
I’d really prefer not to use wildcard selectors if possible

**Comment by lainsce** [comment]
In this case, it's needed because you wouldn't know the actual target name used by third-party apps like Firefox to display their selection, and so on (Hint: it'd bloat the files to get every single selection class used by them :grimacing:). I think this is the only wildcard we'd need to have at most. :pray:

**Comment by danrabbit** [comment]
Adwaita doesn't result to using the wildcard selector, so I'm not convinced that's true

**Comment by lainsce** [comment]
Adwaita instead does repetition and adds `&:selected` blocks to each widget. And there's some wacky class names there as well that I never saw, for ex. `.gtkstyle-fallback` with the `&:selected` block in it as well..

**Comment by lainsce** [comment]
Oh ho, found what Adwaita really does:
```
%selected_items {
  background-color: $selected_bg_color;

  @at-root %nobg_selected_items, & {
    color: $selected_fg_color;

    @at-root %selected_items_disabled,
    &:disabled { color: mix($selected_fg_color, $selected_bg_color, 50%); }

    @at-root %selected_items_backdrop,
    &:backdrop {
      color: $backdrop_selected_fg_color;

      &:disabled { color: mix($backdrop_selected_fg_color, $selected_bg_color, 30%); }
    }
  }
}
```

They then extend this to each widget possible.

File: src/widgets/_entries.scss [hunk 1]
```diff
@@ -14,6 +14,14 @@ entry {
         color: #{'@placeholder_text_color'};
     }
 
+    &:selected {
+        @extend %selected_item;
+    }
+
+    text selection {
+        @extend %selected_item;
+    }
```
Comment (danrabbit) [inline src/widgets/_entries.scss:23]: This is not necessary since we already style `selection`

File: src/widgets/_index.scss [hunk 2]
```diff
@@ -18,28 +18,36 @@ messagedialog,
...
     @define-color warning_color @BANANA_100;
 
-    @define-color selected_bg_color #{'alpha(@accent_color_700, 0.3)'};
-    @define-color selected_fg_color #{'@accent_color_100'};
-}
+    @define-color selected_bg_color #{"alpha(@accent_color_700, 0.3)"};
+    @define-color selected_fg_color #{"@accent_color_100"};
+
+    @define-color theme_selected_bg_color #{"alpha(@accent_color_700, 0.3)"};
+    @define-color theme_selected_fg_color #{"@accent_color_100"};
```
Comment (danrabbit) [inline src/widgets/_index.scss:49]: This is pretty redundant. We should just have these point back to `selected_bg_color` and `selected_fg_color`

File: src/widgets/_views.scss [hunk 3]
```diff
@@ -29,3 +39,13 @@ scrolledwindow {
...
 }
+
+textview {
+    text selection {
+        &:focus,
+        & {
+            @extend %selected_item;
+        }
+    }
+}
```
Comment (danrabbit) [inline src/widgets/_views.scss:50]: Unnecessary since we already style `selection`

File: src/widgets/_views.scss [hunk 4]
```diff
@@ -18,6 +18,16 @@ scrolledwindow {
...
 
+    &:selected {
+        &:focus, & {
+            @extend %selected_item;
+        }
+    }
+
+    text selection {
+        @extend %selected_item;
+    }
```
Comment (danrabbit) [inline src/widgets/_views.scss:29]: Unnecessary since we already style `selection`

File: src/_mixins.scss [hunk 5]
```diff
@@ -192,6 +192,16 @@ $toplevel-border-color: rgba(black, 0.2);
     }
 }
 
+%selected_item {
+    @if $color-scheme == "dark" {
+        background-color: #{"@selected_bg_color"};
+        color: #{"@selected_fg_color"};
+    } @else if $color-scheme == "light" {
+        background-color: #{"@selected_bg_color"};
+        color: #{"@selected_fg_color"};
```
Comment (danrabbit) [inline src/_mixins.scss:201]: This does the exact same thing for both, so this makes no sense

**Comment by lainsce** [comment]
All review points solved.

File: src/widgets/_index.scss [hunk 6]
```diff
@@ -18,28 +18,35 @@ messagedialog,
...
+@define-color theme_selected_fg_color #{"@selected_fg_color"};
 @define-color wm_shadow #{rgba(black, 0.2)};
 
-@define-color selected_bg_color #{'alpha(@accent_color_100, 0.3)'};
-@define-color selected_fg_color #{'@accent_color_900'};
+selection {
+    @extend %selected_item;
+}
+
+.gtkstyle-fallback {
```
Comment (danrabbit) [inline src/widgets/_index.scss:36]: Can you make a comment about what this is and where it's used

File: src/widgets/_index.scss [hunk 7]
```diff
@@ -18,28 +18,35 @@ messagedialog,
...
-    @define-color selected_fg_color #{'@accent_color_100'};
-}
+    @define-color selected_bg_color #{"alpha(@accent_color_700, 0.3)"};
+    @define-color selected_fg_color #{"@accent_color_100"};
 
-selection {
-    background-color: #{'@selected_bg_color'};
-    color: #{'@selected_fg_color'};
+    @define-color theme_selected_bg_color #{"@selected_bg_color"};
+    @define-color theme_selected_fg_color #{"@selected_fg_color"};
```
Comment (danrabbit) [inline src/widgets/_index.scss:49]: You shouldn't have to define this twice. You set it to the same thing whether it's dark or not

File: src/widgets/_index.scss [hunk 8]
```diff
@@ -18,28 +18,35 @@ messagedialog,
...
 @if $color-scheme == "dark" {
     @define-color error_color @STRAWBERRY_300;
     @define-color success_color @LIME_300;
     @define-color warning_color @BANANA_100;
 
-    @define-color selected_bg_color #{'alpha(@accent_color_700, 0.3)'};
-    @define-color selected_fg_color #{'@accent_color_100'};
-}
+    @define-color selected_bg_color #{"alpha(@accent_color_700, 0.3)"};
+    @define-color selected_fg_color #{"@accent_color_100"};
```
Comment (danrabbit) [inline src/widgets/_index.scss:46]: Can you undo this change to quotations please

File: src/widgets/_views.scss [hunk 9]
```diff
@@ -18,6 +18,12 @@ scrolledwindow {
     background-color: bg-color(1);
     color: $fg-color;
 
+    &:selected {
+        &:focus, & {
```
Comment (danrabbit) [inline src/widgets/_views.scss:22]: This selector doesn't really make sense to me. Can you explain what you're trying to do?

File: src/widgets/_entries.scss [hunk 10]
```diff
@@ -14,6 +14,10 @@ entry {
         color: #{'@placeholder_text_color'};
     }
 
+    &:selected {
```
Comment (danrabbit) [inline src/widgets/_entries.scss:17]: Can you give an example of when an entry would be selected?

File: src/widgets/_menus.scss [hunk 11]
```diff
@@ -25,6 +25,10 @@
         background-color: #{'alpha(@fg_color, 0.15)'};
     }
 
+    &:selected {
+        @extend %selected_item;
+    }
```
Comment (danrabbit) [inline src/widgets/_menus.scss:30]: Can you give an example of where menuitems or modelbuttons are selected?

File: src/widgets/_index.scss (hunk 6 above)
Comment (lainsce) [reply src/widgets/_index.scss:36]: This is used wherever an item/widget uses GTK colors but isn't GTK proper.
Very useful when dealing with non elementary apps.

File: src/widgets/_index.scss (hunk 7 above)
Comment (lainsce) [reply src/widgets/_index.scss:49]: Will fix.

File: src/widgets/_index.scss (hunk 8 above)
Comment (lainsce) [reply src/widgets/_index.scss:46]: Isn't `"` more correct than `'` in Sass for this thing? The official docs say one uses `"` so I'm confused.

Comment (lainsce) [reply src/widgets/_index.scss:46]: https://sass-lang.com/documentation/interpolation#quoted-strings

Relevant docs

File: src/widgets/_views.scss (hunk 9 above)
Comment (lainsce) [reply src/widgets/_views.scss:22]: If X is X:selected:focus or X:focus, or :selected X.

File: src/widgets/_entries.scss (hunk 10 above)
Comment (lainsce) [reply src/widgets/_entries.scss:17]: By default a Gtk.Entry is selected unless you remove it's focus to another widget at creation, right? This is where this comes into play.

File: src/widgets/_menus.scss (hunk 11 above)
Comment (lainsce) [reply src/widgets/_menus.scss:30]: Some third-party apps have this distinction for coloring. Should I remove this then?

File: src/widgets/_index.scss (hunk 6 above)
Comment (lainsce) [reply src/widgets/_index.scss:36]: In another PR from this, I'll add in non-selected states to it to complete it.

File: src/widgets/_views.scss (hunk 9 above)
Comment (lainsce) [reply src/widgets/_views.scss:22]: I'll rewrite this to be more concise

File: src/widgets/_menus.scss (hunk 11 above)
Comment (lainsce) [reply src/widgets/_menus.scss:30]: Going to remove since it's not used in elementary.

File: src/widgets/_entries.scss (hunk 10 above)
Comment (danrabbit) [reply src/widgets/_entries.scss:17]: You're thinking of :focus

**Comment by lainsce** [comment]
Superceded by advancements in other PRs.
